R/expr-struct.R

```diff
@@ -34,25 +65,103 @@ expr_struct_field <- function(...) {
   })
 }
 
-expr_struct_rename_fields <- function(...) {
+#' Rename the fields of the struct
+#'
+#' @param names New names, given in the same order as the struct's fields.
+#'
+#' @inherit as_polars_expr return
+#' @examples
+#' df <- pl$DataFrame(
+#'   aaa = c(1, 2),
+#'   bbb = c("ab", "cd"),
+#'   ccc = c(TRUE, NA),
+#'   ddd = list(1:2, 3)
+#' )$select(struct_col = pl$struct("aaa", "bbb", "ccc", "ddd"))
+#' df
+#'
+#' df <- df$select(
+#'   pl$col("struct_col")$struct$rename_fields(c("www", "xxx", "yyy", "zzz"))
+#' )
+#' df$select(pl$col("struct_col")$struct$field("*"))
+#'
+#' # Following a rename, the previous field names cannot be referenced:
+#' tryCatch(
+#'   {
+#'     df$select(pl$col("struct_col")$struct$field("aaa"))
+#'   },
+#'   error = function(e) print(e)
+#' )
+expr_struct_rename_fields <- function(names) {
   wrap({
-    check_dots_unnamed()
-
-    dots <- list2(...) |>
-      unlist(recursive = FALSE)
-    check_character(dots, arg = "...", allow_na = FALSE)
-
-    self$`_rexpr`$struct_rename_fields(dots)
```

Comment:
Why?

Reply:
It's not `*` in py-polars, it's a vector: https://docs.pola.rs/api/python/dev/reference/expressions/api/polars.Expr.struct.rename_fields.html

And the input in Rust is `StringSexp` so I don't see why we'd use `...` here

Reply:
I see. In that case, please use the check function to make sure there is no NA.

Reply:
Done